src/NServiceBus.AcceptanceTests/Recoverability/When_handler_defines_recoverability_configuration.cs

```diff
@@ -0,0 +1,152 @@
+
+public class When_handler_defines_recoverability_configuration : NServiceBusAcceptanceTest
+{
+    [Test]
+    public async Task Should_apply_handler_configuration_on_handler_exceptions()
+    {
+        var context = await Scenario.Define<Context>()
+            .WithEndpoint<EndpointWithRecoverabilityHandler>(e => e
+                .DoNotFailOnErrorMessages()
+                .When(s => s.SendLocal(new FailingMessage())))
+            .Done(c => c.FailedMessages.Any())
+            .Run();
+
+        Assert.AreEqual(11, context.HandlerInvocations);
+        //Assert.AreEqual(typeof(FailingMessage), context.ReceivedMessageType);
+    }
+
+
+    [Test]
+    public async Task Should_not_apply_handler_configuration_on_pipeline_exceptions()
+    {
+        var context = await Scenario.Define<Context>()
+            .WithEndpoint<EndpointWithRecoverabilityHandler>(e => e
+                .CustomConfig(c => c.Pipeline.Register(new ExceptionChangingBehavior(), "will catch handler exceptions and rethrow a different one"))
+                .DoNotFailOnErrorMessages()
+                .When(s => s.SendLocal(new FailingMessage())))
+            .Done(c => c.FailedMessages.Any())
+            .Run();
+
+        Assert.AreEqual(1, context.HandlerInvocations);
+    }
+
+    [Test]
+    public async Task Should_apply_handler_configuration_on_pipeline_rethrow()
+    {
+        var context = await Scenario.Define<Context>()
+            .WithEndpoint<EndpointWithRecoverabilityHandler>(e => e
+                .CustomConfig(c => c.Pipeline.Register(new ExceptionRethrowingBehavior(), "rethrows exceptions"))
+                .DoNotFailOnErrorMessages()
+                .When(s => s.SendLocal(new FailingMessage())))
+            .Done(c => c.FailedMessages.Any())
+            .Run();
+
+        Assert.AreEqual(11, context.HandlerInvocations);
+    }
+
+    class Context : ScenarioContext
+    {
+        public int HandlerInvocations { get; set; }
+        public Type ReceivedMessageType { get; set; }
+    }
+
+    class ExceptionChangingBehavior : Behavior<IIncomingPhysicalMessageContext>
+    {
+        public override async Task Invoke(IIncomingPhysicalMessageContext context, Func<Task> next)
+        {
+            try
+            {
+                await next();
+            }
+            catch (Exception e)
+            {
+                throw new Exception("Custom exception that wraps the original exception", e);
+            }
+        }
+    }
+
+    class ExceptionRethrowingBehavior : Behavior<IIncomingPhysicalMessageContext>
+    {
+        public override async Task Invoke(IIncomingPhysicalMessageContext context, Func<Task> next)
+        {
+            try
+            {
+                await next();
+            }
+            catch (Exception e)
+            {
+                // do some logging or whatever
+#pragma warning disable CA2200 // Rethrow to preserve stack details
+                throw e; // rethrow and change stacktrace
```

Comment:
What's the intent here?

Reply:
just want to check that the exception.GetHashCode isn't changed by changing the stack trace